kernel/cpu.c

```diff
@@ -1880,20 +1878,28 @@ static bool __init cpuhp_bringup_cpus_parallel(unsigned int ncpus)
 	return true;
 }
 #else
-static inline bool cpuhp_bringup_cpus_parallel(unsigned int ncpus) { return false; }
+static inline bool cpuhp_bringup_cpus_parallel(const struct cpumask *mask, unsigned int ncpus) { return false; }
 #endif /* CONFIG_HOTPLUG_PARALLEL */
 
-void __init bringup_nonboot_cpus(unsigned int max_cpus)
+void __init bringup_nonboot_cpus(unsigned int max_cpus, const struct cpumask *boot_cpus)
 {
+	const struct cpumask *mask = cpu_present_mask;
+
+	if (boot_cpus) {
+		static struct cpumask tmp_mask __initdata;
+		cpumask_and(&tmp_mask, mask, boot_cpus);
```

Comment:
Can we avoid tmp_mask by:
cpumask_and(mask, mask, boot_cpus);

Reply:
That would be modifying cpu_present_mask right? I don't think we want that.

Reply:
Right. This approach is taken elsewhere and has no effect on binary size (in BSS) or memory use (thanks to __initdata), so I think it's a good one.